### DISCUSSION TITLE
Show latency/ping time on client to tell if someone is going to have an issue

### DISCUSSION THREAD

**Discussion by wandido** [jamulussoftware/jamulus#1060] [UI Related]
We are always asking each other what their latency is especially if the jam is not in sync.

This will make it easy to tell who has high latency issue and make quick decisions.

**Comment by corrados** [comment]
The problem is that the server does not calculate a ping time to each client. Just the clients send ping times. So the information about the ping times are at the clients, not at the server. But if you want to distribute the ping time information, this information must be shipped to the server and then further shipped to all clients.

**Comment by wandido** [comment]
> The problem is that the server does not calculate a ping time to each client. Just the clients send ping times. So the information about the ping times are at the clients, not at the server. But if you want to distribute the ping time information, this information must be shipped to the server and then further shipped to all clients.

That should work perfectly.

**Comment by wandido** [comment]
Send ping time along with **Musician Profile**

**Comment by matthimue** [comment]
This issue is also somehow related to  https://github.com/corrados/jamulus/issues/150#issue-607598175 so may be both issues might be considered together and be merged. We also encountered the problem, to identify which client contributes to dropouts and I requested an indicator so that every one can see, who has an issue sending data to the server, or who has an issue with his systemsetup, due to wrong buffer settings. Adding delay value visible to everyone would be also a great improvement!

**Comment by gilgongo** [comment]
> especially if the jam is not in sync.

@wandido I'm sure you know this, but for the benefit other people reading this who might not:

If people are having trouble staying in sync, it's always best to double-check that nobody is trying to listen to their direct signal before going on to troubleshoot in other ways. Not listening to your own sound coming back from the server will pretty much guarantee you won't be able to play in time with each other.

**Comment by gilgongo** [comment]
This seems related to a number of discussions about telemetry in Jamulus: https://github.com/jamulussoftware/jamulus/discussions/1032

So that we can make the Issues list a clearer set of actionable things for the backlog, I'm moving this (and #150) to a discussion if that's OK.
